### DISCUSSION TITLE
Current supported ECMAScript version of @babel/parser

### DISCUSSION THREAD

**Discussion by matz3** [babel/babel#16571] [Q&A] [answered]
Hey!

I'd like to figure out what the supported ECMAScript version of the @babel/parser package is.
The [documentation states ES2020](https://github.com/babel/website/blob/873907a11bef7a1a1123a140953e858d8e13ff2e/docs/parser.md#L10), which I doubt is up to date.

Thanks!

**Comment by liuxingbaoyu** [comment, marked as answer]
`@babel/parser` should support all stable features.
It doesn't look up to date here, as at least `classProperties` (ES 2022) is enabled by default.

**Reply by matz3** [reply]
Thanks for the quick reply 👍🏻

I have tested the hashbang grammar specifically, as I wanted to make sure that jsdoc (which internally uses `@babel/parser`) supports ES2023. My test succeeded, but that is because `jsdoc` (v4) [preprocesses the source file to comment out hashbang grammar](https://github.com/jsdoc/jsdoc/blob/49b6fc221156ff2d0d9232e07b84ab8303895941/lib/jsdoc/src/parser.js#L79-L80).

When removing that preprocessing, the `@babel/parser` does not fail, but the AST does not include a comment node for the hashbang grammar. Based on that I would agree that right now it likely supports ES2022, but not ES2023 where I would expect the AST to include the hashbang comment. Anyhow, for my scenario this is not relevant.

**Reply by liuxingbaoyu** [reply]
![image](https://github.com/babel/babel/assets/30521560/753ae320-a180-4bc8-94cb-76f0b98c5c3b)
It should be contained in a `Program` node.

**Reply by matz3** [reply]
Oh, I'm sorry I did not check that. Thanks for looking into it!

I just knew that `espree` introduced a new comment type `Hashbang`, so I also expected it to be within the comments in the babel AST and did not look any further.
